sympy/matrices/expressions/matexpr.py

```diff
@@ -97,6 +97,8 @@ def _mul_handler(self):
         return MatMul
 
     def __neg__(self):
+        if isinstance(self, MatAdd):
+            return MatAdd(*[MatMul(S.NegativeOne, mat).doit() for mat in self.args])
         return MatMul(S.NegativeOne, self).doit()
```

Comment:
What @sylee957 meant is that, what L100 and L101 are doing should have been done by `MatMul(S.NegativeOne, self).doit()`. 
```python
In [1]: A = MatrixSymbol('A', m, n)

In [2]: B = MatrixSymbol('B', m, n)

In [3]: from sympy import MatMul

In [4]: c = Symbol('c')

In [5]: mat_expr = A + B

In [6]: MatMul(-c, mat_expr).doit()
Out[6]: -c⋅(A + B)

In [7]: MatMul(-c, mat_expr)
Out[7]: -c⋅(A + B)

In [9]: MatMul(-c, mat_expr).args
Out[9]: (-c, A + B)

In [10]: MatMul(-c, mat_expr).doit(deep=False)
Out[10]: -c⋅(A + B)

In [11]: 

```